tests/compile-fail/enums_clike.rs

```diff
@@ -0,0 +1,53 @@
+#![feature(plugin, associated_consts)]
+#![plugin(clippy)]
+#![deny(clippy)]
+
+#![allow(unused)]
+
+#[repr(usize)]
+enum NonPortable {
+    X = 0x1_0000_0000, //~ ERROR: Clike enum variant discriminant is not portable to 32-bit targets
+    Y = 0,
+    Z = 0x7FFF_FFFF,
+    A = 0xFFFF_FFFF,
+}
+
+enum NonPortableNoHint {
+    X = 0x1_0000_0000, //~ ERROR: Clike enum variant discriminant is not portable to 32-bit targets
```

Comment:
simple, this one is inferred to be `isize`. Don't ask me why, I think it's a bug. https://github.com/rust-lang/rust/blob/master/src/test/run-pass/enum-discrim-autosizing.rs shows that even on 32-bit, the size is 8 bytes but the type is `isize` xD